### PR TITLE
feat(proto):  modifed proto fetcher to generate files in place

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**

### CODE REVIEW

File: src/interfaces/IMassaProtoFile.ts [hunk 1]
```diff
@@ -5,7 +5,8 @@
  * @see candidate_value - If it is different from the final_value, it represents the
  * candidate file (as a byte array) to replace the current proto file.
  */
-export interface IProtoFile {
-  final_value: Uint8Array;
-  candidate_value: Uint8Array;
+export interface IMassaProtoFile {
```
Comment (gregLibert) [inline src/interfaces/IMassaProtoFile.ts:8]: Can you remove the useless I prefix for the interface?
If there is more, can you create an issue about that ?

File: src/web3/SmartContractsClient.ts [hunk 2]
```diff
@@ -498,10 +501,25 @@ export class SmartContractsClient
...
-
+      let protoFiles: IMassaProtoFile[];
       // parse response
       const json = await response.json();
-      return json.result;
+      for (let proto of json) {
+        let content = proto['final_value'].toString();
+        let protos = content.split('syntax = "proto3";'); // splitting all the proto functions to make separate proto file for each functions
+        for (let func of protos) {
+          const rName = /message (.+)RHelper/gm;
```
Comment (gregLibert) [inline src/web3/SmartContractsClient.ts:511]: This regexp is dangerous and can behave differently based on the regex engine or configuration if the initial function contains RHelper in its name.
You should use the fact that the function name is follow by a white space.

```suggestion
          const rName = /message (.+)RHelper /gm;
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Comment by gregLibert** [comment]
@0xSwapFeeder the Ci is broken.

**Comment by 0xSwapFeeder** [comment]
> @0xSwapFeeder the Ci is broken.

Done

**Pull request merged**